### ISSUE TITLE
add documentation on using weights with labelPropagation

### ISSUE THREAD

**Issue by jexp** [neo4j-contrib/neo4j-graph-algorithms#420]
which supports both node-weights and relationship-weights

see:  

```
            double partition = partition(targetNodeId);
            double weight = graph.weightOf(sourceNodeId, targetNodeId) * graph.weightOf(targetNodeId);
            votes.addTo(partition, weight);
```

and other algorithms that support weights.

**Comment by tomasonjo** [comment]
I think [#270](https://github.com/neo4j-contrib/neo4j-graph-algorithms/issues/270) might still be the cause, that LPA does not function properly. I wasn't really sure if node-weights are meant to be initial partition label, that could be used as semi-supervised data cleansing or such. I tried it both on 3.2.x and 3.1.x

Test data:
```
CREATE (nAlice:User {id:'Alice',label:2})
,(nBridget:User {id:'Bridget',label:3})
,(nCharles:User {id:'Charles',label:4})
,(nDoug:User {id:'Doug',label:3})
,(nMark:User {id:'Mark',label: 4})
,(nMichael:User {id:'Michael',label:2})
CREATE (nAlice)-[:FOLLOW]->(nBridget)
,(nAlice)-[:FOLLOW]->(nCharles)
,(nMark)-[:FOLLOW]->(nDoug)
,(nBridget)-[:FOLLOW]->(nMichael)
,(nDoug)-[:FOLLOW]->(nMark)
,(nMichael)-[:FOLLOW]->(nAlice)
,(nAlice)-[:FOLLOW]->(nMichael)
,(nBridget)-[:FOLLOW]->(nAlice)
,(nMichael)-[:FOLLOW]->(nBridget)
,(nCharles)-[:FOLLOW]->(nDoug)
```
If we dont use an existing partitionProperty only 4 nodes end up with partition set, where all ( 6 ) should have it.

```
CALL algo.labelPropagation('User', 'FOLLOW','OUTGOING', {iterations:10,partitionProperty:'partition', write:true}) 
YIELD nodes, iterations, loadMillis, computeMillis, writeMillis
```
Using existing partitionProperty as an semi-supervised algo feels like how it is supposed to be used, but suffers from above problem aswell.

```
CALL algo.labelPropagation('User', 'FOLLOW','OUTGOING', {iterations:10,partitionProperty:'label', write:true}) 
YIELD nodes, iterations, loadMillis, computeMillis, writeMillis
```
Played around with it a bit and it is acting very strange. First time you run the algo it load 3, 4 or 5 nodes randomly, which make different results. The second time you run it on the same graph it always loads 0 nodes. One explanation for this is that the labels have already converged and it skips loading or sth. Anyhow it is very confusing for a user.
@mknblch

**Comment by knutwalker** [comment]
I fixed #270 in #432.
@tomasonjo I implemented some changes that are described in https://arxiv.org/pdf/0709.2938.pdf to reduce oscillations, but LPA is not quite deterministic, randomness is desired. LPA now produces much more stable results than before, but might still occasionally produce different cluster.
For example, the group of `Charles`, `Doug`, and `Mark` here might end up in a single cluster with any of those nodes as their label or `Doug` and `Mark` might end up switching to each others label, depending on batch size, parallelism as in thread execution order, local node execution order.
